etna/core/models/documents.py

```diff
@@ -0,0 +1,23 @@
+from django.db import models
+
+from wagtail.documents.models import AbstractDocument, Document
+
+
+class CustomDocument(AbstractDocument):
+    extent = models.CharField(blank=True, null=True)
+    description = models.TextField(blank=True, null=True)
+
+    @property
+    def pretty_file_size(self):
+        suffixes = ["B", "KB", "MB", "GB"]
+        i = 0
+        pretty_file_size = self.file_size
+        while pretty_file_size >= 1000 and i < len(suffixes) - 1:
+            pretty_file_size /= 1000
+            i += 1
+        return f"{pretty_file_size:.1f}{suffixes[i]}"
```

Comment:
What would be great here is a different number of decimal places for different sizes.

For B and kB we don't really need any decimal places but as the file gets larger, those decimal places get more important.

Example:

- `50B` (0 DP for B)
- `50kB` (0 DP for kB)
- `50.1MB` (1 DP for MB)
- `5.12GB` (2 DP for GB)

Another thing that would be nice is not to end with a `.0` - e.g. `50.0MB` should ideally just be `50MB`.

Reply:
Something like this would take care of the decimal points:

```python
return f"{round(file_size, min(suffix_index - 1, 2))}{suffixes[suffix_index]}"
```

Not sure how is best to remove the `.0` or `.00` if there is one.